### PR TITLE
WIP: Custom metrics docs

### PR DESCRIPTION
**Pull request by paigecalvert**
https://deploy-preview-1408--replicated-docs.netlify.app/vendor/custom-metrics

### CODE REVIEW

File: docs/vendor/custom-metrics.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+`Authorization` header is required for this API and must contain license ID.
+
+The location of the API endpoint is different depending on if KOTS or the SDK is installed in the cluster:
+* For applications installed with KOTS, the in-cluster API custom metrics endpoint is located at `http://kotsadm:3000/api/v1/metrics`. 
+
+  **Example:**
+
+  ```
+  POST http://kotsadm:3000/api/v1/metrics
```
Comment (sgalsaleh) [inline docs/vendor/custom-metrics.md:72]:
```suggestion
  POST http://kotsadm:3000/api/v1/app/custom-metrics
```

File: docs/vendor/custom-metrics.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  The instance detail will show `Active Users: 10` and `Using Custom Reports: false`, which represents the most recent payload received. The previously-sent `numProjects` value is discarded from the instance summary and is available in the instance events payload.
+
+## Configure Custom Metrics
+
+You can configure your application to send a set of metrics as key value pairs to the API that is running in the cluster alongside the application instance.
+
+`Authorization` header is required for this API and must contain license ID.
+
+The location of the API endpoint is different depending on if KOTS or the SDK is installed in the cluster:
+* For applications installed with KOTS, the in-cluster API custom metrics endpoint is located at `http://kotsadm:3000/api/v1/metrics`.
```
Comment (sgalsaleh) [inline docs/vendor/custom-metrics.md:67]:
```suggestion
* For applications installed with KOTS, the in-cluster API custom metrics endpoint is located at `http://kotsadm:3000/api/v1/app/custom-metrics`. 
```

File: docs/vendor/custom-metrics.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  ```json
+  {
+    "data": {
+      "num_projects": 5,
+      "weekly_active_users": 10
+    }
+  }
+  ```
+
+* For Helm chart-based applications that include the Replicated SDK, the in-cluster API custom metrics endpoint is located at `http://replicated-sdk:3000/api/v1/metrics`.
```
Comment (sgalsaleh) [inline docs/vendor/custom-metrics.md:84]:
```suggestion
* For Helm chart-based applications that include the Replicated SDK, the in-cluster API custom metrics endpoint is located at `http://replicated-sdk:3000/api/v1/app/custom-metrics`.
```

File: docs/vendor/custom-metrics.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+  **Example:**
+
+  ```bash
+  POST http://replicated-sdk:3000/api/v1/metrics
```
Comment (sgalsaleh) [inline docs/vendor/custom-metrics.md:89]:
```suggestion
  POST http://replicated-sdk:3000/api/v1/app/custom-metrics
```

File: docs/vendor/custom-metrics.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    const projectsQuery = "SELECT COUNT(*) as num_projects from projects";
+    const numProjects = (await db.getConnection().queryOne(projectsQuery)).num_projects;
+
+    const usersQuery = 
+        "SELECT COUNT(*) as active_users from users where DATEDIFF('day', last_active, CURRENT_TIMESTAMP) < 7";
+    const activeUsers = (await db.getConnection().queryOne(usersQuery)).active_users;
+
+    const metrics = { data: { numProjects, activeUsers }};
+    
+    await fetch('https://replicated-sdk:3000/api/v1/metrics', {
```
Comment (sgalsaleh) [inline docs/vendor/custom-metrics.md:117]:
```suggestion
    await fetch('https://replicated-sdk:3000/api/v1/app/custom-metrics', {
```

File: docs/vendor/custom-metrics.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+          - name: hello
+            image: postgres:14
+            command:
+            - /bin/sh
+            - -c
+            - |
+                date; echo sending metrics
+                activeUsers=$(psql -t -c 'select COUNT(*) from active_users')
+                numProjects=$(psql -t -c 'select COUNT(*) from projects')                
+                curl -X POST https://replicated-sdk:3000/api/v1/metrics -H 'Authorization: ${licenseId}' --data-binary "{\"activeUsers\":${activeUsers}, \"numProjects\":${numProjects}}"
```
Comment (sgalsaleh) [inline docs/vendor/custom-metrics.md:171]:
```suggestion
                curl -X POST https://replicated-sdk:3000/api/v1/app/custom-metrics -H 'Authorization: ${licenseId}' --data-binary "{\"activeUsers\":${activeUsers}, \"numProjects\":${numProjects}}"
```

File: static/images/custom-metrics-diagram.png
Comment (sgalsaleh) [inline static/images/custom-metrics-diagram.png:1]: wrong endpoint in the screenshot, should be `/api/v1/app/custom-metrics`

File: docs/vendor/custom-metrics.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+The following diagram demonstrates how a custom `activeUsers` metric is sent to the in-cluster API and ultimately displayed in the vendor portal, as described above:
+
+![Custom metrics flowing from customer environment to vendor portal](/images/custom-metrics-diagram.png)
+[View a larger version of this image](/images/custom-metrics-diagram.png)
+
+## Requirements
+
+To support the collection of custom metrics, one of the following must be running in the cluster alongside the application instance:
+* Replicated KOTS version 1.112.0 or later
+* Replicated SDK version 0.0.1-beta.9 or later
```
Comment (sgalsaleh) [inline docs/vendor/custom-metrics.md:26]:
```suggestion
* Replicated SDK version 0.0.1-beta.5 or later
```

File: docs/vendor/custom-metrics.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+The vendor portal collects custom metrics through Replicated KOTS or through the Replicated SDK, depending on which is installed in the cluster alongside the application instance. KOTS and the SDK both expose an in-cluster API where you can configure your application to POST metric payloads. When an application instance sends data to the API, KOTS or the SDK sends the data (including any custom and built-in metrics) to the Replicated app service. The app service is located at `replicated.app` or at your custom domain.
+
+If any values in the metric payload are different from the current values for the instance, then a new event is generated and displayed in the vendor portal. For more information about how the vendor portal generates events, see [How the Vendor Portal Generates Events and Insights](/vendor/instance-insights-event-data#how-the-vendor-portal-generates-events-and-insights) in _About Instance and Event Data_.
+
+## Requirements
+
+To support the collection of custom metrics, one of the following must be running in the cluster alongside the application instance:
+* Replicated KOTS version 1.112.0 or later
```
Comment (divolgin) [inline docs/vendor/custom-metrics.md:20]:
```suggestion
* Replicated KOTS version [TBD, not yet available]
```

File: docs/vendor/custom-metrics.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
+# Configuring Custom Metrics (Alpha)
+
+This topic describes how to configure an application to send custom metrics to the Replicated vendor portal.
+
+:::note
+The custom metrics feature is Alpha and is subject change. To access this feature, open a feature request.
```
Comment (divolgin) [inline docs/vendor/custom-metrics.md:6]:
```suggestion
The custom metrics feature is Alpha and is subject change. To access this feature, open a [feature request](https://vendor.replicated.com/support).
```

**Review by AmberAlston — APPROVED** [review]

**Comment by AmberAlston** [comment]
Note for VenEx team and for @paigecalvert - We needed to merge this to support the SDK release. When KOTS is released we'll update with the respective KOTS version. We will need to add back the helpful diagram but had to remove it temporarily as the endpoint shown was wrong and we didn't have the ability to update it quickly.

**Pull request merged**